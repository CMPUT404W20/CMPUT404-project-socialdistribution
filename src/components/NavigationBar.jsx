import React, { Component } from "react";
import Fade from "react-reveal/Fade";
import "../styles/NavigationBar.scss";
import HomeOutlinedIcon from "@material-ui/icons/HomeOutlined";
import PeopleAltOutlinedIcon from "@material-ui/icons/PeopleAltOutlined";
import NotificationsNoneOutlinedIcon from "@material-ui/icons/NotificationsNoneOutlined";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import PropTypes from "prop-types";
import InputGroup from "react-bootstrap/InputGroup";
import FormControl from "react-bootstrap/FormControl";
import NavDropdown from "react-bootstrap/NavDropdown";
import { NavLink, withRouter } from "react-router-dom";
import logo from "../images/logo.svg";
import * as auth from "../services/AuthenticationService";

class NavigationBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: localStorage.getItem("username") || "Username",
      userID: localStorage.getItem("userID"),
      numNotifications: 2,
      keyword: "",
    };
  }

  handleLogOut = () => {
    auth.logoutUser().then((response) => {
      if (response.status === 200) {
        localStorage.clear();
      }
    });
  }

  handleSearchTextChange = (event) => {
    this.setState({ keyword: event.target.value });
  }

  handleSearchKeyPress = (event) => {
    const { keyword } = this.state;
    const { history } = this.props;
    if (event.key === "Enter") {
      event.preventDefault();
      if (keyword !== "") {
        history.push(`/search?username=${keyword}`);
      }
    }
  }

  render() {
    const {
      username, userID, numNotifications, keyword,
    } = this.state;
    return (
      <Navbar collapseOnSelect expand="sm" fixed="top" className="navigationBar">
        <Navbar.Brand className="logo">
          <img src={logo} width="85%" alt="app logo" />
        </Navbar.Brand>
        <div className="search-input-container">
          <InputGroup size="sm" className="searchBar" onSubmit={this.handleRedirect}>
            <FormControl
              placeholder="Search"
              aria-label="Search"
              aria-describedby="basic-addon1"
              value={keyword}
              onChange={this.handleSearchTextChange}
              onKeyPress={this.handleSearchKeyPress}
            />
          </InputGroup>
        </div>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="mr-auto" />
          <Nav>
            <Nav.Link exact as={NavLink} to="/home">
              <HomeOutlinedIcon />
            </Nav.Link>
            <Nav.Link exact as={NavLink} to="/friends">
              <PeopleAltOutlinedIcon />
            </Nav.Link>
            <Nav.Link exact as={NavLink} to="/notifications">
              <div className="notification-icon-wrapper">
                <NotificationsNoneOutlinedIcon />
                <div className="notification-badge-wrapper">
                  <span className="notification-badge">{numNotifications}</span>
                </div>
              </div>
            </Nav.Link>
          </Nav>
          <Nav>
            <NavDropdown title={username} id="username-dropdown" alignRight>
              <Fade left duration={500} distance="5px">
                <NavDropdown.Item
                  as={NavLink}
                  exact
                  to={{
                    pathname: `/profile/${username}`,
                    state: { userID, username },
                  }}
                >
                  Profile
                </NavDropdown.Item>
                <NavDropdown.Item href="#">Settings</NavDropdown.Item>
                <NavDropdown.Divider />
                <NavDropdown.Item as={NavLink} exact to="/" onSelect={this.handleLogOut}>Logout</NavDropdown.Item>
              </Fade>
            </NavDropdown>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}

NavigationBar.propTypes = {
  history: PropTypes.objectOf(PropTypes.checkPropTypes()).isRequired,
};

export default withRouter(NavigationBar);
