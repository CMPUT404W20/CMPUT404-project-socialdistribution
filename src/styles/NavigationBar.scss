@import "./common.scss";
$badge-bg-color: #F24B4B;
.navigationBar {
    background-color: white;
    box-shadow: 0px 0px 10px rgba(121, 121, 121, 0.25);
    ;
    width: 100%;
    .search-input-container {
        width: 50%;
        .searchBar {
            max-width: 400px;
            .form-control {
                border: 1px white solid;
            }
            .form-control:focus {
                border: 1px $border-color solid;
                box-shadow: unset;
            }
        }
    }
    .navbar-brand {
        margin-right: unset;
    }
    .logo {
        font-weight: 700;
        font-size: 18px;
    }
    .nav-link {
        padding-right: 0 !important;
        padding-left: 35px !important;
        font-size: 10pt;
        text-align: center;
        font-weight: 500;
        line-height: 10px;
        transition: color 0.1s ease-in;
        &.active {
            color: $accent-color !important;
        }
        &:hover {
            color: $secondary-color !important;
        }
        .notification-icon-wrapper {
            display: flex;
            justify-content: center;
            .notification-badge-wrapper {
                background: $badge-bg-color;
                height: 10pt;
                min-width: 11pt; // if there are a lot of notifications (ex: 1000), we want this badge to expand to fit the number
                border-radius: 50%;
                margin-left: -0.6rem;
                .notification-badge {
                    font-weight: 700;
                    color: white !important;
                    font-size: 6pt;
                    padding: 2px;
                }
            }
        }
    }
    .navbar-toggler {
        border: none !important;
        padding: 0 !important;
        font-size: 14px !important;
        outline: 0 !important;
    }
    .dropdown-menu {
        @extend %dropdown;
    }
    .dropdown-toggle::after {
        border: none;
    }
    @media screen and (max-width: 576px) {
        .nav-link {
            padding-left: 0px !important;
        }
        .form-control {
            text-align: center;
        }
    }
}

::-webkit-input-placeholder,
::-moz-placeholder {
    color: $placeholder-text-color;
    font-style: normal;
    font-weight: normal;
}