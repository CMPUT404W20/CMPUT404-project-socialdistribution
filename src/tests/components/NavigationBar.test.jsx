import React from "react";
import { shallow } from "enzyme";
import NavigationBar from "../../components/NavigationBar";

describe("Navigation bar Component", () => {
  it("should render correctly", () => {
    const component = shallow(<NavigationBar />);
    expect(component).toMatchSnapshot();
  });
});
