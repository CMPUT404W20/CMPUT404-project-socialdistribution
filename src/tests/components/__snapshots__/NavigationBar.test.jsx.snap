// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar Component should render correctly 1`] = `
<Navbar
  className="navigationBar"
  collapseOnSelect={true}
  expand="sm"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    className="logo"
  >
    <img
      alt="app logo"
      src="logo.svg"
      width="85%"
    />
  </NavbarBrand>
  <div
    className="search-input-container"
  >
    <InputGroup
      className="searchBar"
      size="sm"
    >
      <FormControl
        aria-describedby="basic-addon1"
        aria-label="Search"
        placeholder="Search"
      />
    </InputGroup>
  </div>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    />
    <Nav
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "NavLink",
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        disabled={false}
        exact={true}
        to="/home"
      >
        <HomeOutlinedIcon />
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "NavLink",
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        disabled={false}
        exact={true}
        to="/friends"
      >
        <PeopleAltOutlinedIcon />
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "NavLink",
            "propTypes": Object {
              "activeClassName": [Function],
              "activeStyle": [Function],
              "aria-current": [Function],
              "className": [Function],
              "exact": [Function],
              "innerRef": [Function],
              "isActive": [Function],
              "location": [Function],
              "onClick": [Function],
              "replace": [Function],
              "strict": [Function],
              "style": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        disabled={false}
        exact={true}
        to="/notifications"
      >
        <div
          className="notification-icon-wrapper"
        >
          <NotificationsNoneOutlinedIcon />
          <div
            className="notification-badge-wrapper"
          >
            <span
              className="notification-badge"
            >
              2
            </span>
          </div>
        </div>
      </NavLink>
    </Nav>
    <Nav
      fill={false}
      justify={false}
    >
      <NavDropdown
        alignRight={true}
        id="username-dropdown"
        title="Username"
      >
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          disabled={false}
          exact={true}
          to={
            Object {
              "pathname": "/profile/Username",
              "state": Object {
                "userID": null,
                "username": "Username",
              },
            }
          }
        >
          Profile
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
          href="#"
        >
          Settings
        </DropdownItem>
        <DropdownDivider
          role="separator"
        />
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "NavLink",
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          disabled={false}
          exact={true}
          onSelect={[Function]}
          to="/"
        >
          Logout
        </DropdownItem>
      </NavDropdown>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;
